kbchat/kbchat_test.go

```diff
@@ -0,0 +1,119 @@
+package kbchat
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+)
+
+type Bots struct {
+	Alice1   *OneshotOptions
+	Alice2   *OneshotOptions
+	Bob1     *OneshotOptions
+	Charlie1 *OneshotOptions
+}
+
+type Config struct {
+	Bots
+}
+
+func readAndParseConfig() (Config, error) {
+	var config Config
+	data, err := ioutil.ReadFile("test_config.yaml")
+	if err != nil {
+		return Config{}, err
+	}
+
+	err = yaml.Unmarshal(data, &config)
+	if err != nil {
+		return Config{}, err
+	}
+
+	return config, nil
+}
+
+func randomTempDir() (string, error) {
```

Comment:
It's in `/tmp/` so not a huge deal but would probably be good to delete this directory once you're done.